app/src/main/java/com/example/tshlib/DashboardFragment.kt

```diff
@@ -1,14 +1,22 @@
 package com.example.tshlib
 
+import android.content.Intent
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Button
 import androidx.fragment.app.Fragment
 
 class DashboardFragment: Fragment(R.layout.fragment_dashboard) {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        view.findViewById<Button>(R.id.take_quiz_btn).setOnClickListener {
+            val intent = Intent(activity, QuizStartingPage::class.java)
+            activity?.startActivity(intent)
+        }
```

Comment:
The way we have the app set up is by using fragments and replacing the fragment that is being shown, so do you think you can switch to using fragments instead of an activity to keep it consistent? you can refer to the `MainActivity.kt` for that code or lmk if you have any questions.